### PR TITLE
[GR-35277] Fix merging of debug info subranges and make debuginfotest more resiliant to JDK changes

### PR DESCRIPTION
**Pull request by zakkak**
* Stops merging leaf subranges with non-leaf ones.

  As we are generating line info only for leaf ranges, merging a leaf range with a non-leaf one results in not generating line info for the former (previously leaf) range.

* Substitutes or removes tests relying on standard library PrintStream.

**Note**: This PR is based on 4550c69898bf6c0eec11e4965f86b66b25396268 instead of `master` to avoid the AssertionError introduced by https://github.com/oracle/graal/commit/ee638aadb6201aa713e44c4865858a84f856d06e and reported in #4018 so that it can be tested without getting the AssertionError.

Closes: #4036

### CODE REVIEW

**Comment by zakkak** [comment]
@adinn please review (especially https://github.com/oracle/graal/pull/4046/commits/436b06eeffb5c0384d2e39c41d0bb216629e5be6)

File: substratevm/mx.substratevm/testhello.py [hunk 1]
```diff
@@ -541,7 +530,8 @@ def test():
     checker = Checker('ptype hello.Hello', rexp)
     checker.check(exec_string, skip_fails=False)
 
-    # list methods matching regural expression "nlined", inline methods are not listed
+    # list methods matching regural expression "nline", inline methods are not listed because they lack a definition
+    # (this is true for C/C++ as wwell)
```
Comment (olpaw) [inline substratevm/mx.substratevm/testhello.py:505]: s/wwell/well

File: substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/debugentry/Range.java [hunk 2]
```diff
@@ -306,6 +306,10 @@ private Range maybeMergeSibling(DebugContext debugContext) {
             /* cannot merge callers with different line numbers, move on. */
             return sibling;
         }
+        if ( (!isLeaf() && sibling.isLeaf()) || (isLeaf() && !sibling.isLeaf()) ) {
```
Comment (olpaw) [inline substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/debugentry/Range.java:309]: How about:
```java
if (isLeaf() ^ sibling.isLeaf()) {
```

Comment (olpaw) [reply substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/debugentry/Range.java:309]: Or:
```java
if (isLeaf() != sibling.isLeaf()) {
```

Comment (zakkak) [reply substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/debugentry/Range.java]: Oops, that slept away. Thanks Paul!

**Review by olpaw — APPROVED** [review]

**Comment by zakkak** [comment]
@olpaw how can I instruct eclipseformat to skip some lines (or a file)?

It suggests the following change which is not desirable.
```diff
     @AlwaysInline("For testing purposes")
     private static void inlineFrom() {
-        noInlineHere(5); inlineTailRecursion(1); // These two calls are purposely placed on the same line!!!
+        noInlineHere(5);
+        inlineTailRecursion(1); // These two calls are purposely placed on the same line!!!
         inlineHere(5);
         inlineTailRecursion(5);
```

**Comment by olpaw** [comment]
> @olpaw how can I instruct eclipseformat to skip some lines (or a file)?
> 
> It suggests the following change which is not desirable.
> 
> ```diff
>      @AlwaysInline("For testing purposes")
>      private static void inlineFrom() {
> -        noInlineHere(5); inlineTailRecursion(1); // These two calls are purposely placed on the same line!!!
> +        noInlineHere(5);
> +        inlineTailRecursion(1); // These two calls are purposely placed on the same line!!!
>          inlineHere(5);
>          inlineTailRecursion(5);
> ```

use:
```java
//@formatter:off
noInlineHere(5); inlineTailRecursion(1); // These two calls are purposely placed on the same line!!!
//@formatter:on
```

**Comment by zakkak** [comment]
Thank you @olpaw.

I think we should backport this to 21.3 as well. WDYT?

**Pull request merged**